pkg/policyfilter/state.go

```diff
@@ -222,6 +226,25 @@ func (pol *policy) podInfoMatches(pod *podInfo) bool {
 	return pol.podMatches(pod.namespace, pod.labels)
 }
 
+func (pol *policy) containerMatches(container *containerInfo) bool {
+	containerFilterFields := labels.Labels{
+		"name": container.name,
+	}
+	return pol.containerSelector.Match(containerFilterFields)
```

Comment:
Maybe I'm missing something but isn't the case that `containerSelector` might be nil? What would happen then?

Reply:
You are right, it can be `nil` when we read it from the tracing policy.
However, [before](https://github.com/cilium/tetragon/pull/2231/commits/7aebeacc6e6ef7473b5b8e2423eea72f071892f7#diff-980e5be14cfb8e087d0d09a0f42b7219a9510ad91dd199221abdc98d1767bf34R428) adding it to a new `policy`, we construct the selector from label selector and in case the `containerSelector` is `nil`, this will be an empty selector `[]selectorOp{}`.
[Here](https://github.com/cilium/tetragon/blob/main/pkg/labels/labels.go#L71-L72)'s the corresponding code. Also, the [same logic](https://github.com/cilium/tetragon/blob/main/pkg/policyfilter/state.go#L398) is used for `podSelector`.

Reply:
Thanks!
Am I correct to assume that there is a test for the above path (i.e, being `nil` in the tracing policy and having an empty selector when we call this function)? I would think yes, since AFAIR we do have tests where the container selector is nil.

Reply:
Yes, you are right, I tried to test the behavior when `containerSelector` is `nil` in multiple places!
- For example, [here](https://github.com/cilium/tetragon/pull/2231/commits/4764f1bb530d10166fc16c20e582a49d71b1d848#) I test that the policy with the `nil` pod and container selectors matches all containers.
- I also added the nil `containerSelector` to existing tests to make sure that the policy with the `podSelector` but without `containerSelector` still produces the same results. For example, [here](https://github.com/cilium/tetragon/pull/2231/commits/4764f1bb530d10166fc16c20e582a49d71b1d848#diff-0dfcd52d313187c0d7956a105e745763c25de1309923b1f8db9d3450af33363fR413).
- Also, just in case, [here](https://github.com/cilium/tetragon/pull/2231/commits/4764f1bb530d10166fc16c20e582a49d71b1d848#diff-0dfcd52d313187c0d7956a105e745763c25de1309923b1f8db9d3450af33363fR497) I tested that the policy with the `containerSelector` but without `podSelector` also matches all specified containers.

I tried to test other scenarios too, for example, when pod labels/containers change and we re-add containers. Hopefully, I covered the majority of cases :pray: